compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1725,7 +1725,21 @@ impl<'a> Parser<'a> {
                 err.help("see https://doc.rust-lang.org/book/ch05-03-method-syntax.html for more information");
                 err
             } else {
-                self.expected_ident_found()
+                let mut err = self.expected_ident_found();
+                if let Some((ident, _)) = self.token.ident() {
```

Comment:
You also need something like the following (I think that `ident` won't coerce to `&str`, so this exact code won't work, but I think there's an `as_str()` method):

```suggestion
                if let Some((ident, _)) = self.token.ident() && ident == "let" {
```